lib/pure/os.nim

```diff
@@ -75,8 +75,8 @@ elif defined(posix):
 else:
   {.error: "OS module not ported to your operating system!".}
 
-when weirdTarget and defined(nimErrorProcCanHaveBody):
-  {.pragma: noNimScript, error: "this proc is not available on the NimScript target".}
```

Comment:
I spent effort on this specialized error message, why remove it?

Reply:
this was already answered here: https://github.com/nim-lang/Nim/pull/14143#issuecomment-620484977

Reply:
> because the problems in #14142 are real and outweigh the benefits of having a nicer error message.

Well that's debatable and I don't agree. It's simply true that some os procs are unavailable for NimScript and a clear error message helps.

Reply:
addressed in latest commit